synapse/lib/rstorm.py

```diff
@@ -74,6 +75,37 @@ async def _mockHttp(self, *args, **kwargs):
 
         return s_stormhttp.HttpResp(info)
 
+    @contextlib.contextmanager
+    def _shimHttpCalls(self, vcr_kwargs):
+        path = self.ctx.get('mock-http-path')
+        if not vcr_kwargs:
+            vcr_kwargs = {}
+
+        if path:
+            path = os.path.abspath(path)
+            # try it as json first (since yaml can load json...). if it parses, we're old school
+            # if it doesn't, either it doesn't exist/we can't read it/we can't parse it.
+            # in any of those cases, default to using vcr
+            try:
+                with open(path, 'r') as fd:
+                    byts = json.load(fd)
+            except (FileNotFoundError, json.decoder.JSONDecodeError):
+                byts = None
+
+            if not byts:
+                with vcr.use_cassette(os.path.abspath(path), **vcr_kwargs) as cass:
+                    yield cass
```

Comment:
It is worth it even yield this? Maybe for a future version?

Reply:
I dont see a need, but it's also harmless :D